index.bs

```diff
@@ -560,79 +560,251 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+			1. For each (|key|, |value|) in |items|:
+
+				1. Let |mimeType| be the result of [=parsing a MIME type=] given |key|.
+
+				1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+				1. If [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=] [=list/contains=] a [=representation=] whose [=representation/MIME type=] is |mimeType|, then throw a {{TypeError}}. 
+
+				1. Let |representation| be a new [=representation=].
+
+				1. Set |representation|'s [=representation/MIME type=] to |mimeType|.
+
+				1. Set |representation|'s [=representation/data=] to |value|.
+
+				1. Append |representation| to [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
+
+				1. Let |mimeTypeString| be the result of [=serializing a MIME type=] with |mimeType|.
+				
+				1. Add |mimeTypeString| to |types|.
+
+			1. Set [=this=]'s [=ClipboardItem/types array=] to the result of running [=create a frozen array=] from |types|.
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			The {{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</h4>
+		
+			This method must run the below steps:
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parsing a MIME type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Let |p| be [=a new promise=] in |realm|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=representation/MIME type=] is |mimeType|, then:
+
+					1. Let |representationDataPromise| be the |representation|'s [=representation/data=].
+
+					1. [=promise/React=] to |representationDataPromise|:
+
+						1. If |representationDataPromise| was fulfilled with value |v|, then:
+
+							1. If |v| is a {{DOMString}}, then follow the below steps:
+
+								1. Let |dataAsBytes| be the result of [=UTF-8 encoding=] |v|.
+
+								1. Let |blobData| be a {{Blob}} created using |dataAsBytes| with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
+
+								1. Resolve |p| with |blobData|.
+
+							1. If |v| is a {{Blob}}, then follow the below steps:
+
+								1. Resolve |p| with |v|.
+
+						1. If |representationDataPromise| was rejected, then:
+
+							1. [=Reject=] |p| with {{"NotFoundError"}} {{DOMException}} in |realm|.
```

Comment:
In response to https://github.com/w3c/clipboard-apis/pull/158/#discussion_r777750170: it seems reasonable to add something like
```
Issue: web-developers might be interested in the underlying rejection reason.
```
Not more, to limit the scope of this review.

Reply:
Ack.